mouse/mouse.nim

```diff
@@ -42,113 +51,82 @@ when defined(windows):
             Release: int
 
     const
-        Left*: ButtonTuple = (MOUSEEVENTF_LEFTDOWN, MOUSEEVENTF_LEFTUP)
-        Right*: ButtonTuple = (MOUSEEVENTF_RIGHTDOWN, MOUSEEVENTF_RIGHTUP)
-        Midle*: ButtonTuple = (MOUSEEVENTF_MIDDLEDOWN, MOUSEEVENTF_MIDDLEUP)
+        MouseButtonToEvent: array[MouseButton, ButtonTuple] = [
+            Left: (MOUSEEVENTF_LEFTDOWN, MOUSEEVENTF_LEFTUP),
+            Right: (MOUSEEVENTF_RIGHTDOWN, MOUSEEVENTF_RIGHTUP),
+            Middle: (MOUSEEVENTF_MIDDLEDOWN, MOUSEEVENTF_MIDDLEUP)
+        ]
         X*: ButtonTuple = (MOUSEEVENTF_XDOWN, MOUSEEVENTF_XUP)
 
-        Horizontal*: int = MOUSEEVENTF_HWHEEL
-        Vertical*: int = MOUSEEVENTF_WHEEL
+        ScrollOrientationToEvent = [
+            Horizontal: MOUSEEVENTF_HWHEEL,
+            Vertical: MOUSEEVENTF_WHEEL
+        ]
 
-        Relative*: string = "relative"
-        Absolute*: string = "absolute"
 
 
+    proc click*(button: MouseButton, x = 0, y = 0) =
+        mouse_event(DWORD(MouseButtonToEvent[button].Press), DWORD(x), DWORD(y), 0, 0)
+        mouse_event(DWORD(MouseButtonToEvent[button].Release), DWORD(x), DWORD(y), 0, 0)
 
+    proc press*(button: MouseButton, x = 0, y = 0) =
+        mouse_event(DWORD(MouseButtonToEvent[button].Press), DWORD(x), DWORD(y), 0, 0)
 
-    proc click*(button: ButtonTuple, x = 0, y = 0) =
-        mouse_event(DWORD(button.Press), DWORD(x), DWORD(y), 0, 0)
-        mouse_event(DWORD(button.Release), DWORD(x), DWORD(y), 0, 0)
+    proc release*(button: MouseButton, x = 0, y = 0) =
+        mouse_event(DWORD(MouseButtonToEvent[button].Release), DWORD(x), DWORD(y), 0, 0)
 
-    proc press*(button: ButtonTuple, x = 0, y = 0) =
-        mouse_event(DWORD(button.Press), DWORD(x), DWORD(y), 0, 0)
-
-    proc release*(button: ButtonTuple, x = 0, y = 0) =
-        mouse_event(DWORD(button.Release), DWORD(x), DWORD(y), 0, 0)
-
-    proc move*(x: int32, y: int32, `type` = Relative) =
-        if `type` == "relative":
+    proc move*(x, y: int, typ = Relative) =
```

Comment:
Not sure why this accepted a string - an enum is much better, and allows for shorter code.